projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -420,27 +384,34 @@ public function add_tools_menu( $wp_admin_import = false, $wp_admin_export = fal
 
 	/**
 	 * Adds Settings menu.
-	 *
-	 * @param bool $wp_admin Optional. Whether links should point to Calypso or wp-admin. Default false (Calypso).
 	 */
-	public function add_options_menu( $wp_admin = false ) {
+	public function add_options_menu() {
+		$submenus_to_update = array();
+
 		$this->hide_submenu_page( 'options-general.php', 'sharing' );
 
-		// There is not complete feature parity between WP Admin and Calypso settings https://github.com/Automattic/wp-calypso/issues/51189.
-		$this->update_submenus( 'options-general.php', array( 'options-general.php' => 'https://wordpress.com/settings/general/' . $this->domain ) );
-		add_submenu_page( 'options-general.php', esc_attr__( 'Advanced General', 'jetpack' ), __( 'Advanced General', 'jetpack' ), 'manage_options', 'options-general.php', null, 1 );
+		// When no preferred view has been set for "Settings > General", keep the previous behavior that forced the
+		// default view regardless of the global preference.
+		$preferred_view = $this->get_preferred_view( 'options-general.php', true );
+		if ( self::DEFAULT_VIEW === $preferred_view || self::UNKNOWN_VIEW === $preferred_view ) {
+			$submenus_to_update['options-general.php'] = 'https://wordpress.com/settings/general/' . $this->domain;
+		}
 
-		add_submenu_page( 'options-general.php', esc_attr__( 'Performance', 'jetpack' ), __( 'Performance', 'jetpack' ), 'manage_options', 'https://wordpress.com/settings/performance/' . $this->domain, null, 2 );
+		if ( self::DEFAULT_VIEW === $this->get_preferred_view( 'options-writing.php' ) ) {
+			$this->hide_submenu_page( 'options-general.php', 'options-writing.php' );
+		}
 
-		if ( $wp_admin ) {
-			return;
+		if ( self::DEFAULT_VIEW === $this->get_preferred_view( 'options-discussion.php' ) ) {
+			$this->hide_submenu_page( 'options-general.php', 'options-discussion.php' );
```

Comment:
Now that we have distinct Calypso pages we should aim at updating these submenus rather than hiding them
`https://wordpress.com/settings/discussion/`

Reply:
Good catch, seems I made a mistake while solving some merge conflicts. Fixed in https://github.com/Automattic/jetpack/pull/20147/commits/e0c066e20a76d8faa64451f1dd2ddcaf95810a8c